core/src/main/java/io/questdb/cutlass/text/TextImportTask.java

```diff
@@ -0,0 +1,995 @@
+                    lexer.setTableName(tableNameSink);
+                    lexer.of(columnDelimiter);
+                    lexer.setSkipLinesWithExtraValues(false);
+                    try {
+                        for (int i = lo; i < hi; i++) {
+                            lexer.clear();
+                            errors = 0;
+                            final CharSequence name = partitions.get(i).name;
+                            path.of(importRoot).concat(name);
+                            mergePartitionIndexAndImportData(ff, path, lexer);
+
+                            long imported = atomicity == Atomicity.SKIP_ROW ? lexer.getLineCount() - errors : lexer.getLineCount();
+                            importedRows.add(i);
+                            importedRows.add(imported);
+
+                            LOG.info().$("Imported data [partition=").$(name).$(",lines=").$(lexer.getLineCount()).$(",errors=").$(errors).$("]").$();
+                        }
+                    } finally {
+                        writer.commit(CommitMode.SYNC);
+                    }
+                }
+            }
+        }
+
+        public LongList getImportedRows() {
+            return importedRows;
+        }
+
+        public void clear() {
+            this.cairoEngine = null;
+            this.targetTableStructure = null;
+            this.types = null;
+            this.atomicity = -1;
+            this.columnDelimiter = (byte) -1;
+            this.importRoot = null;
+            this.inputFileName = null;
+            this.index = -1;
+            this.partitions = null;
+            this.timestampIndex = -1;
+            this.timestampAdapter = null;
+
+            this.errors = 0;
+            this.offset = 0;
+            this.importedRows.clear();
+            this.tableNameSink.clear();
+        }
+
+        private void logError(long offset, int column, final DirectByteCharSequence dbcs) {
+            LOG.error().$("type syntax [type=").$(ColumnType.nameOf(types.getQuick(column).getType())).$(",line offset=").$(offset).$(",column=").$(column).$(" value='").$(dbcs).$("']").$();
+        }
+
+        private void importPartitionData(final TextLexer lexer, long address, long size) throws TextException {
+            final CairoConfiguration configuration = cairoEngine.getConfiguration();
+            final FilesFacade ff = configuration.getFilesFacade();
+
+            long bufLength = configuration.getSqlCopyBufferSize();
+            long buf = Unsafe.malloc(bufLength, MemoryTag.NATIVE_DEFAULT);
+            long fd = -1;
+            int utf8SinkSize = cairoEngine.getConfiguration().getTextConfiguration().getUtf8SinkSize();
+            try (Path path = new Path().of(configuration.getInputRoot()).concat(inputFileName);
+                 DirectCharSink utf8Sink = new DirectCharSink(utf8SinkSize)) {
+                fd = ff.openRO(path.$());
```

Comment:
should use `TableUtils.openRO()` 

this code is also untested for file open failure